Sprint 09

ArtemDubovitskiy opened the pull request.

ArtemDubovitskiy committed:
Добавлен экран профиля пользователя

ArtemDubovitskiy committed:
Добавлен контейнер UITabBarController, белый статус бар

ArtemDubovitskiy committed:
Добавлен UINavigationController

ArtemDubovitskiy committed:
Создан SingleImageViewController для просмотра одной картинки. Добавлен переход с экрана с таблицей (ImageListView) на новый экран.

ArtemDubovitskiy committed:
Настроен сегвей от таблицы к просмотру одной картинки

ArtemDubovitskiy committed:
Изменен переход с таблицы на отдельное изображения с Navigation Bar на модальное окно

ArtemDubovitskiy committed:
Добавлен ScrollView на экран просмотра одного изображения

ArtemDubovitskiy committed:
Реализована функциональность зума. Добавлена кнопка "Поделиться".

ArtemDubovitskiy committed:
Добавлен файл с перечнем статических названий цветов по макету

ArtemDubovitskiy committed:
Экран профиля пользователя сверстан кодом.

LizaIlinykh commented on ImageFeed-iOS/ImagesList/ImagesListViewController.swift:
```diff
@@ -19,15 +17,31 @@ final class ImagesListViewController: UIViewController {
+    override func prepare(for seque: UIStoryboardSegue, sender: Any?) {
+        if seque.identifier == ShowSingleImageSegueIdentifier {
+            let viewController = seque.destination as! SingleImageViewController
+            let indexPath = sender as! IndexPath
```
давай уберем принудительные касты заменив их на optional binding

ArtemDubovitskiy replied:
получается этот код я могу написать как?:
```swift
guard let viewController = seque.destination as? SingleImageViewController else {
                return
            }
            guard let indexPath = sender as? IndexPath else {
                return
            }
```

LizaIlinykh replied:
лучше так
```
    override func prepare(for seque: UIStoryboardSegue, sender: Any?) {
        guard seque.identifier == ShowSingleImageSegueIdentifier,
              let viewController = seque.destination as? SingleImageViewController,
              let indexPath = sender as? IndexPath else {
            super.prepare(for: seque, sender: sender)
            return
        }
        let imageName = photosName[indexPath.row]
        let image = UIImage(named: "\(imageName)_full_size") ?? UIImage(named: imageName)
        viewController.image = image
    }
```

ArtemDubovitskiy replied:
Спасибо! поправлю

LizaIlinykh commented on ImageFeed-iOS/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,95 @@
+import UIKit
+
+final class ProfileViewController: UIViewController {
+//    private var label: UILabel?
```
если закомментированный код не нужен, лучше его удалять

ArtemDubovitskiy replied:
спасибо, пропустил

LizaIlinykh commented on ImageFeed-iOS/SingleImage/SingleImageViewController.swift:
```diff
@@ -0,0 +1,62 @@
+import UIKit
+
+final class SingleImageViewController: UIViewController {
+    var image: UIImage! {
```
давай здесь избавимся от force типа , сделаем просто опционал, либо можно просто задать тип UIImage и задать начальное значение

ArtemDubovitskiy replied:
+

LizaIlinykh replied:
ты убрал принудительное извлечение , но перенес его в другие места. давай сделаем так например
`rescaleAndCenterImageInScrollView(image: image ?? UIImage())`

или так
```
    var image: UIImage? {
        didSet {
            guard isViewLoaded,
                  let image else { return }
            imageView.image = image
            rescaleAndCenterImageInScrollView(image: image)
        }
    }
```

ArtemDubovitskiy replied:
Спасибо!

LizaIlinykh commented on ImageFeed-iOS/Profile/ProfileViewController.swift:
```diff
@@ -0,0 +1,95 @@
+    
+    private let logoutButton: UIButton = {
+        let button = UIButton.systemButton(
+            with: UIImage(systemName: "ipad.and.arrow.forward")!,
```
давай здесь тоже избавимся от принудительного извлечения

ArtemDubovitskiy committed:
Исправления по замечаниям ревьюера.
Системное изображение logoutButton заменил на .png из макета с фиксированными размерами.

LizaIlinykh commented on ImageFeed-iOS/ImagesList/ImagesListViewController.swift:
```diff
@@ -27,8 +27,14 @@ final class ImagesListViewController: UIViewController {
     // MARK: - PrepareForSegue
```
проще прощения, что сразу не обратила твое внимание
ты скачиваешь большую иконку логаута из фигмы, а нужно скачать маленькую и задать ей четкие размеры 20x22
я не могу отправить картинки на платформе, поэтому здесь приложу(ты скачиваешь ту которая 44X44, а нужно 20X22)
<img width="100" alt="Снимок экрана 2023-07-27 в 18 25 18" src="https://github.com/ArtemDubovitskiy/ImageFeed-iOS/assets/140049788/6fbe1786-9619-477f-b1a8-27fc8e2da197">
<img width="109" alt="Снимок экрана 2023-07-27 в 18 25 38" src="https://github.com/ArtemDubovitskiy/ImageFeed-iOS/assets/140049788/6fa66cef-3597-4862-b2ac-c31f653c117a">

ArtemDubovitskiy replied:
Спасибо, я понял. Я экспортировал картинку с областью нажатия)
Поправлю этот момент сейчас.

ArtemDubovitskiy replied:
Сделал кнопку с отступами как в макете.

ArtemDubovitskiy committed:
Исправления по замечаниям ревьюера: загружено изображение 20х22.
Область нажатия кнопки 44х44, изображение в кнопке расположено согласно макету.

ArtemDubovitskiy committed:
Исправлен крем при нажатии на кнопку logoutButton
